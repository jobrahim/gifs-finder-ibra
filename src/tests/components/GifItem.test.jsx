import { render, screen } from "@testing-library/react"
import { GifItem } from "../../components";


describe('Prueba de GiftItem', () => { 

  const title = 'Saitama';
  const url = 'http://one-puns.lskd.com/'
    test('debe hacer match con el snapshot', () => { 
      const {container} = render(<GifItem title={title} url={url}/>)
      expect(container).toMatchSnapshot();

     })

     test('debe mostrar la imagen con el url y el alt indicado', () => { 
      render(<GifItem title={title} url={url}/>);
      //expect(screen.getByRole('img').src).toBe(url);

      const { src , alt} = screen.getByRole('img');
      expect(src).toBe(url);
      expect(alt).toBe(alt);
      
     })

     test('debe mostrar el titulo en el componente', () => { 
      render(<GifItem title={title} url={url}/>);
      expect(screen.getByText(title)).toBeTruthy;

      
     })
 })