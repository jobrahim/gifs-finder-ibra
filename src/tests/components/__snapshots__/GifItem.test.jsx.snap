// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GiftItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="http://one-puns.lskd.com/"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
